perf/lua/column_scan_module.c

```diff
@@ -97,7 +98,51 @@ sum_iterator_rv_lua_func(struct lua_State *L)
 }
 #endif /* defined(ENABLE_READ_VIEW) */
 
-#if defined(ENABLE_SCANNER)
+#if defined(ENABLE_ARROW_AND_SCANNER)
+static int
+sum_arrow_lua_func(struct lua_State *L)
+{
+	uint32_t space_id = luaL_checkinteger(L, 1);
+	uint32_t index_id = luaL_checkinteger(L, 2);
+	uint32_t field_no = luaL_checkinteger(L, 3);
+	char key[8];
+	char *key_end = mp_encode_array(key, 0);
+	uint32_t fields[] = {field_no};
+	uint32_t field_count = lengthof(fields);
+	box_arrow_options_t *options = box_arrow_options_new();
+	box_arrow_options_set_batch_row_count(options, 4096);
+	struct ArrowArrayStream *stream = xmalloc(sizeof(*stream));
+	int rc = box_index_arrow_stream(space_id, index_id, field_count, fields,
+					key, key_end, options, stream);
+	if (rc != 0) {
+		free(stream);
+		box_arrow_options_delete(options);
+		return luaT_error(L);
+	}
+	uint64_t sum = 0;
+	struct ArrowArray array;
+	while (true) {
+		rc = stream->get_next(stream, &array);
+		if (rc != 0 || array.n_children != 1)
+			break;
+		uint64_t *values = (uint64_t *)array.children[0]->buffers[1];
+		for (int i = 0; i < (int)array.children[0]->length; i++)
+			sum += values[i];
+		if (array.release != NULL)
+			array.release(&array);
+	}
+	stream->release(stream);
+	free(stream);
+	box_arrow_options_delete(options);
+	if (rc != 0) {
+		if (array.release != NULL)
+			array.release(&array);
```

Comment:
I think we should release the array not only on error but also on eof (if `array.n_children == 1`). Same comment to the read view version.

Reply:
Fixed.